### PR TITLE
Fix kwargs not passed to SubCenterArcFaceLoss

### PR DESCRIPTION
**Pull request by marcpaga**
`**kwargs` are not passed into `__init__` of `SubCenterArcFaceLoss`.

This leads to problems, for example with `reducers`, which have no effect when passed.

Example bug:
```
arcfaceloss = SubCenterArcFaceLoss(
    num_classes=10,
    sub_centers=3,
    embedding_size=64,
    reducer=DoNothingReducer(),
)

emb = torch.randn(4, 64)
print(arcfaceloss(emb, torch.arange(4)))

# without fix, we get the mean of the losses, which is the default reducer, despite passing `DoNothingReducer` to the `init`.
> tensor(39.9293, grad_fn=<MeanBackward0>)

# with fix, the reducer is properly passed and is properly applied.
> {'loss': {'losses': tensor([53.9860, 42.8221, 38.8233, 62.1092], grad_fn=<MulBackward0>), 'indices': tensor([0, 1, 2, 3]), 'reduction_type': 'element'}}
```

The fix includes the use of `deepcopy` to ensure no reference problems occur when deleting the keys form `**kwargs`, which is necessary otherwise duplicate keys are passed into `__init__`. But perhaps it can be done more elegantly.

### CODE REVIEW

**Comment by KevinMusgrave** [comment]
Thanks! Seems reasonable. Can you add your example to the tests here: https://github.com/KevinMusgrave/pytorch-metric-learning/blob/master/tests/losses/test_subcenter_arcface_loss.py

Something like this should be sufficient I think:

```python
from pytorch_metric_learning.reducers import DoNothingReducer


def test_reducer(self):
  arcfaceloss = SubCenterArcFaceLoss(
      num_classes=10,
      sub_centers=3,
      embedding_size=64,
      reducer=DoNothingReducer(),
  )
  
  emb = torch.randn(4, 64)
  result = arcfaceloss(emb, torch.arange(4))
  self.assertTrue(isinstance(result, dict))
```

**Comment by marcpaga** [comment]
Hi @KevinMusgrave,
I have added the unittest, with also a shape check based on the batch dim.
Cheers!

**Comment by KevinMusgrave** [comment]
Thanks!
